api/src/main/java/javax/enterprise/inject/spi/AfterBeanDiscovery.java

```diff
@@ -64,6 +67,21 @@
     public void addBean(Bean<?> bean);
 
     /**
+     *
+     * Obtains a new {@link BeanConfigurator} to configure  a new {@link Bean}  and add it at the end of the observer invocation.
+     * It will then fire an event of type {@link javax.enterprise.inject.spi.ProcessBean} containing the built
+     * {@link javax.enterprise.inject.spi.Bean} from this configuration and then register it with the
+     * container, thereby making it available for injection into other beans.
+     *
+     * Each call returns a new BeanConfigurator.
+     *
+     * @return a non reusable {@link BeanConfigurator} to configure the bean to add
+     * @throws IllegalStateException if called outside of the observer method invocation
+     * @since 2.0
+     */
+    public BeanConfigurator<?> addBean();
```

Comment:
This method must be parameterized `<T> BeanConfigurator<T> addBean()`. Otherwise we wouldn't be able to pass lamdas to create callbacks. The same applies to `addObserverMethod()` and other `add()` methods.